Engine/EffectInstanceRenderRoI.cpp

```diff
@@ -737,17 +737,21 @@ EffectInstance::renderRoI(const RenderRoIArgs & args,
         //renderRoIInternal should check the bitmap of 'image' and not downscaledImage!
         roi = args.roi.toNewMipmapLevel(args.mipmapLevel, 0, par, rod);
 
-        if (frameArgs->tilesSupported && !roi.clipIfOverlaps(upscaledImageBoundsNc)) {
-            return eRenderRoIRetCodeOk;
+        if (frameArgs->tilesSupported) {
+            if (!roi.clipIfOverlaps(upscaledImageBoundsNc)) {
+                return eRenderRoIRetCodeOk;
+            }
+            assert(upscaledImageBoundsNc.contains(roi));
         }
-        assert( upscaledImageBoundsNc.contains(roi));
     } else {
         roi = args.roi;
 
-        if (frameArgs->tilesSupported && !roi.clipIfOverlaps(downscaledImageBoundsNc)) {
-            return eRenderRoIRetCodeOk;
+        if (frameArgs->tilesSupported) {
+            if (!roi.clipIfOverlaps(downscaledImageBoundsNc)) {
```

Comment:
Could we avoid side effects in conditions, and either have a const function overlaps(), and do the clip outside of the condition?
I think this clipIfOverlaps() function should be removed from the code, because it is used as a test at many places but also has a dangerous and misleading side effect.
No one expects a condition to modify the variable being tested. See c++ core guideline es.40 https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es40-avoid-complicated-expressions

Reply:
Friendly ping. I'd like to land this and #929 so the CI builds start working again. I'm happy to follow up with clipIfOverlap() fixes once the build and tests have been restored.